dumpling/export/dump.go

```diff
@@ -706,6 +737,65 @@ func (d *Dumper) sequentialDumpTable(tctx *tcontext.Context, conn *BaseConn, met
 	return d.dumpWholeTableDirectly(tctx, meta, taskChan, "", orderByClause, 0, 1)
 }
 
+func (d *Dumper) handleBuffer(totalChunk int) {
+	if !d.conf.SQLConcurrent {
+		return
+	}
+	set := make(map[int]*bytes.Buffer)
+	cond := sync.NewCond(new(sync.Mutex))
+	finished := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-finished:
+				return
+			case r := <-d.bufChan:
+				cond.L.Lock()
+				set[r.idx] = r.buf
+				cond.Signal()
```

Comment:
I don't think signal is a good way to collaborate with routines. It's an outmoded fashion from C. I consider channel would be a more modern and advanced version of signal. Can we just try using channel here (e.g., map[int]channel) since it looks more succinct and intuitive.

Reply:
I don't think channel is good at situations here. Maintaining a channels map whose length is `len(chunks)` costs more and looks more difficult to me. We still need a lock for map set. I think `sync.Cond` looks more clear and costs less. It's good at this flow control case.

Reply:
Nit: `signal` can be simply replaced with a single channel, e.g  `signal chan struct{}`, no need to maintain a channel map.

Reply:
maybe golang's channel is hard to prevent "(producer) insert to map and notify" from happening between "(consumer) check chunk is not ready" and "(consumer) wait". consumer can't lock the map mutation and waiting as a whole without condition variable.

Reply:
How about this? https://github.com/pingcap/tidb/pull/40553/commits/5271a7d617447ba7364fa62bd3ad3c172e8738e2